woocommerce/wc-admin
# Migrate header items to slot fills

Fixes (part of) #7689 

Migrates header items to slot fills to keep feature logic separate and allow header extensibility.

Follow-ups/thoughts:

* We need to export the header slots for use externally.  GB has very specific packages for each area, but this may be overkill for our use case (i.e. `@woocommerce/header`).  Depending on how much we anticipate adding, we could do create a `@woocommerce/layout` package.  Up for discussion.
* It seems that SlotFill does not support any ordering 😱 .  This PR uses the flexbox `order` property to organize items, but I really wish there was an official API or method for sorting these fills.
* The back button could have more generalized use, but it does not at the moment and is specific to tasks so I've moved it there.  I did make the component a bit more generic though.  Maybe this should be moved back into the header and we should create a `@woocommerce/header` package?

### Screenshots

![Screen Shot 2021-10-15 at 9 21 51 AM](https://user-images.githubusercontent.com/10561050/137495482-a7579189-0238-43ea-afd4-d848a83d3b74.png)

### Detailed test instructions:

1. Navigate to various pages and make sure the activity panel works as expected and still shows the correct items.
2. Enable the navigation feature under WooCommerce->Settings->Advanced->Features.
3. Navigate to embed and non-embed pages, checking that the navigation works as expected.
4. Navigate into various tasks and make sure the task title is shown correctly and that the back button is rendered in the header.
5. Either spoof your browser as Android or change the `platform() === ANDROID_PLATFORM` manually to `true`.
6. Delete the `android_app_banner_dismissed` user meta option.
7. Navigate to any WCA page and narrow the browser to <782px, making sure the banner is shown, positioned correctly, and works as expected.
